source/slang/glsl.meta.slang

```diff
@@ -833,29 +895,39 @@ public T faceforward(T n, T i, T ng)
 //
 
 __generic<T : __BuiltinFloatingPointType, let C : int, let R : int>
-__target_intrinsic(glsl)
 [__readNone]
 [ForceInline]
 [OverloadRank(15)]
+[require(cpp_cuda_glsl_hlsl_spirv, GLSL_400)]
 public matrix<T, C, R> outerProduct(vector<T, C> c, vector<T, R> r)
 {
-    // Column major matrix in GLSL
-    matrix<T, C, R> result;
-    for (int i = 0; i < C; ++i)
+    __target_switch
     {
-        for (int j = 0; j < R; ++j)
+    case glsl: __intrinsic_asm "outerProduct";
+    default:
+        // Column major matrix in GLSL
+        matrix<T, C, R> result;
+        for (int i = 0; i < C; ++i)
         {
-            result[i][j] = c[i] * r[j];
+            for (int j = 0; j < R; ++j)
+            {
+                result[i][j] = c[i] * r[j];
+            }
         }
+        return result;
     }
-    return result;
 }
 
 __generic<T : __BuiltinFloatingPointType, let N : int>
-__target_intrinsic(hlsl)
-__target_intrinsic(glsl)
 [require(cpp_cuda_glsl_hlsl_spirv, GLSL_400)]
-matrix<T,N,N> inverse(matrix<T,N,N> m);
+matrix<T,N,N> inverse(matrix<T,N,N> m)
+{
+    __target_switch
+    {
+    case glsl: __intrinsic_asm "inverse";
+    case hlsl: __intrinsic_asm "inverse";
+    }
+}
```

Comment:
does this function compile?
it targets cpp, cuda, glsl, hlsl, spirv, but only implements glsl & hlsl.

Reply:
CI is all passing and it compile fine on my local machine, which seems strange.

Reply:
I see in `intrinsic-basic.slang` the test for `inverse` is never ran (`#ifdef`)
```
#if defined(TEST_when_inverse_works)
        && equals(mat2(0), inverse(mat2(0)))
        && equals(mat3(0), inverse(mat3(0)))
        && equals(mat4(0), inverse(mat4(0)))
#endif // #if defined(TEST_when_inverse_works)
```
I believe this may be why no error appears?